File: talker_listener_int/scripts/listener_int.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import Int32

def callback(data):
    rospy.loginfo(data)

def listener_int():
    rospy.init_node('listener_int_py', anonymous=True)
    rospy.Subscriber('int_chatter', Int32, callback)

    rospy.spin()

if __name__ == "__main__":
    listener_int()

File: talker_listener_int/src/listener_int.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"

void callback(std_msgs::Int32 msg)
{
 ROS_INFO("data is : [%d]", msg.data);
}

int main(int argc, char **argv)
{
 ros::init(argc, argv, "listener_int");
 ros::NodeHandle n;
 
 ros::Subscriber sub = n.subscribe("int_chatter", 1000, callback);
 ros::spin();
 
 return 0;
}

